packages/browser-data-collector/src/report.ts

```diff
@@ -1,48 +1,62 @@
 /**
  * Internal dependencies
  */
-import { getNavigationStart } from './api/performance-timing';
-import { deviceMemory, performanceTiming, environment, networkInformation } from './collectors';
+import {
+	deviceMemory,
+	performanceTiming,
+	environment,
+	networkInformation,
+	fullPageStart,
+	inlineStart,
+	pageVisibilityStart,
+	pageVisibilityStop,
+} from './collectors';
 
 export class ReportImpl implements Report {
 	id: string;
-	start!: number;
+	beginning!: number;
 	end?: number;
-	data: ReportData;
-	collectors!: Collector[];
+	data: ReportData = new Map();
+	startCollectors!: Collector[];
+	stopCollectors!: Collector[];
 
-	constructor( id: string, isInitial: boolean ) {
+	static async fromNow( id: string ) {
+		const report = new ReportImpl( id, false );
+		await report.start();
+		return report;
+	}
+
+	static async fromPageStart( id: string ) {
+		const report = new ReportImpl( id, true );
+		await report.start();
+		return report;
+	}
+
+	private constructor( id: string, isInitial: boolean ) {
```

Comment:
Good usage of private 👍 Can always be made public if we ever want to allow for reports to be directly instanced by others.